Upstream: yes, taken from master

From 0a3b0ee07f9ca932b2e36cb213252aca7394b673 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=96mer=20Fad=C4=B1l=20Usta?= <omerusta@gmail.com>
Date: Tue, 22 Jul 2014 01:30:15 +0300
Subject: [PATCH] Missing fclose - Resource leaking

config resource already opened on the other hand forgotten to close

REVIEW: 119352
---
 startkde/kstartupconfig/kstartupconfig.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/startkde/kstartupconfig/kstartupconfig.cpp b/startkde/kstartupconfig/kstartupconfig.cpp
index f0b902a..d65ee7d 100644
--- a/startkde/kstartupconfig/kstartupconfig.cpp
+++ b/startkde/kstartupconfig/kstartupconfig.cpp
@@ -99,7 +99,10 @@ int main()
     filename = kdehome + "/share/config/startupconfigkeys";
     keys = fopen(filename.c_str(), "r");
     if( keys == NULL )
+    {
+        fclose(config);
         return 2;
+    }
     for(;;)
         {
         char* nl;
-- 
2.0.2

